src/sentry/migrations/0440_remove_unattached_notification_settings.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 2.2.28 on 2023-05-12 16:54
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+NUMBER_RECORDS_APPROX = 11310
+
+
+def remove_unattached_notification_settings(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    query_set = NotificationSetting.objects.filter(user_id__isnull=True, team_id__isnull=True)
+    if len(query_set) <= NUMBER_RECORDS_APPROX:
+        query_set.delete()
```

Comment:
If there are more records this would silently fail. Is that what you're hoping for?

Reply:
The idea is to not delete more records than expected. I can verify via SQL afterwards if the deletes happened.